src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const item of actions) {
+    const stateCopy = { ...currentState };
+
+    if (item.type === 'addProperties') {
+      Object.assign(stateCopy, { ...item.extraData });
+    }
+
+    if (item.type === 'removeProperties') {
+      for (const el of item.keysToRemove) {
+        delete stateCopy[el];
+      }
+    }
+
+    if (item.type === 'clear') {
+      for (const param in stateCopy) {
+        delete stateCopy[param];
```

Comment:
Instead of using a for-in loop to delete all properties, you can simplify the 'clear' action by assigning an empty object to `stateCopy`. This is more efficient and clearer in intent.